### ISSUE TITLE
line codec not working?

### ISSUE THREAD

**Issue by splashx** [logstash-plugins/logstash-codec-line#4]
In a perfectly working environment I've added just one line in the logstash conf input:file section:

```
input {
file {
        path => [  "/mnt/data/*.log"  ]
        start_position => "beginning"
        exclude => [ "*.gz", "*.enc"  ]
        type => "some-type"
        codec => line { charset => "CP1252" }     # <---- this line
   }
}
```

After adding adding this line I see there is nothing being passed to filter workers / output - nothing is heating the ES cluster and ```nload``` in the dst interface shows activity.

After some time logstash crashes with the following message on ```logstash.log```:
```
Exception in thread "<file" java.lang.UnsupportedOperationException
    at java.lang.Thread.stop(Thread.java:869)
    at org.jruby.RubyThread.exceptionRaised(RubyThread.java:1221)
    at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:112)
    at java.lang.Thread.run(Thread.java:745)
```
If I pass ```-w 1 --debug``` and limit ```LS_HEAP_SIZE``` to ```512m``` also nothing comes out of logstash - the log file grows very fast (couple of seconds) to ~ 554MB and after that the growth is very slow (apron. 5 minutes to grow to 559MB) - because the process reached the heap space limit most likely.

I haven't waited long enough but I'm sure logstash will crash after some time.

**Comment by splashx** [comment]
Update: the process indeed crashed after some minutes but there is no error message - also nothing has been sent from logtash :-1:

**Comment by talevy** [comment]
@splashx thank you for reporting this. I know it was a while back, but have you resolved this issue?

It is typical that the Logstash process run out of memory as it grows the String it is occumulating that is meant to represent the "line". If no new line character is ever found, this buffer will grow until it runs out of memory. I believe this is what we saw here

Closing this in favor of #2 for tracking wiser usage of memory while buffering.
